src/services/codefixes/correctQualifiedNameToIndexedAccessType.ts

```diff
@@ -0,0 +1,27 @@
+/* @internal */
+namespace ts.codefix {
+    registerCodeFix({
+        errorCodes: [Diagnostics.Cannot_access_0_1_because_0_is_a_type_but_not_a_namespace_Did_you_mean_to_retrieve_the_type_of_the_property_1_in_0_with_0_1.code],
+        getCodeActions: (context: CodeFixContext) => {
+            const sourceFile = context.sourceFile;
+            const token = getTokenAtPosition(sourceFile, context.span.start, /*includeJsDocComment*/ false);
+            const qualifiedName = getAncestor(token, SyntaxKind.QualifiedName) as QualifiedName;
+            Debug.assert(!!qualifiedName, "Expected position to be owned by a qualified name.");
+            if (!isIdentifier(qualifiedName.left)) {

```

Comment:
This example doesn't seem to trigger the right error, though we can apply the fix manually to fix the issue:
```
module M {
    export interface I {
         foo: string;
    }
}
let a: M.I.foo;
```
Please add a test.